src/ageml/modelling.py

```diff
@@ -206,26 +209,22 @@ def set_hyperparameter_grid(self):
             dict: dictionary with the hyperparameter grid
         """
         param_grid = {}
-        conditions = [self.model_type in AgeML.model_dict.keys(),
-                      self.hyperparameter_tuning > 0,
-                      not self.model_type == "hyperopt"]
+        conditions = [self.model_type in AgeML.model_dict.keys(), self.hyperparameter_tuning > 0, self.model_type != "hyperopt"]
         if all(conditions):
-            hyperparam_ranges = AgeML.model_hyperparameter_ranges[self.model_type]
             hyperparam_types = AgeML.model_hyperparameter_types[self.model_type]
+            invalid_hyperparams = [param for param in self.hyperparameter_params.keys() if param not in hyperparam_types.keys()]
+            if len(invalid_hyperparams) > 0:
+                raise ValueError(f"Hyperparameter(s) {invalid_hyperparams} not available for the selected model '{self.model_type}'.")
+
             # Initialize output grid
-            for hyperparam_name in list(hyperparam_types.keys()):
-                bounds = hyperparam_ranges[hyperparam_name]
-                if hyperparam_types[hyperparam_name] == 'log':
-                    param_grid[f"model__{hyperparam_name}"] = np.logspace(bounds[0],
-                                                                          bounds[1],
-                                                                          int(self.hyperparameter_tuning))
-                elif hyperparam_types[hyperparam_name] == 'int':
-                    param_grid[f"model__{hyperparam_name}"] = np.rint(np.linspace(bounds[0], bounds[1],
-                                                                      int(self.hyperparameter_tuning))).astype(int)
-                elif hyperparam_types[hyperparam_name] == 'float':
-                    param_grid[f"model__{hyperparam_name}"] = np.linspace(bounds[0],
-                                                                          bounds[1],
-                                                                          int(self.hyperparameter_tuning))
+            for hyperparam_name in list(self.hyperparameter_params.keys()):
+                low, high = self.hyperparameter_params[hyperparam_name]
+                if hyperparam_types[hyperparam_name] == "log":
+                    param_grid[f"model__{hyperparam_name}"] = np.logspace(low, high, int(self.hyperparameter_tuning))
```

Comment:
Where are you telling the susers what are logs, ints etc...? Like if I choose C=-3,3 how do i know if it is using log? Or should I have to write C=0.001, 100?

Reply:
I will add it in the documentation. And a link to the documentation explaining this.